### PR TITLE
fix: analyze command default backend bug

### LINKED ISSUE #902: [Bug]: analyze`s --backend arg error
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've included steps to reproduce the behavior

### Affected Components

- [X] K8sGPT (CLI)
- [ ] K8sGPT Operator

### K8sGPT Version

_No response_

### Kubernetes Version

_No response_

### Host OS and its Version

_No response_

### Steps to reproduce

2 problems.

1. set default backend as none openai and give -b openai -> use default backend not openai
```
> ./bin/k8sgpt analyze -f Service -n default -c -e -b openai
   0% |                                                                                                                                                                                                                                                                                | (0/8, 0 it/hr) [0s:0s]
Error: failed while calling AI provider localai: error, status code: 404, message: %!s(<nil>)
```

2. set any default backend and run analyze with -b arg -> only use default backend not provided arg. (i think this is bug)
```
> ./bin/k8sgpt auth default -p noopai
Default provider set to noopai
> ./bin/k8sgpt analyze -f Service -n default -e -c localai
 100% |███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| (7/7, 2693 it/s)
AI Provider: noopai
```

### Expected behaviour

ai provider should be provided `--backend`  even `openai` is given.

### Actual behaviour

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by VaibhavMalik4187**
Closes #902

## 📑 Description
Now, the default value of the `backend` flag for the analyze command will be an empty string. The `NewAnalysis` function has been modified to use the default backend set by the user if the backend flag is not provided and the `defaultprovider` is set in the config file. Otherwise, the backend will be set to "openai".

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Comment by VaibhavMalik4187** [comment]
@JuHyung-Son could you please go through this PR whenever you have some time? Thanks!

**Comment by AlexsJones** [comment]
This looks like it's just moving the functionality elsewhere?

File: pkg/analysis/analysis.go [hunk 1]
```diff
@@ -125,10 +125,14 @@ func NewAnalysis(
 
 	// Backend string will have high priority than a default provider
 	// Backend as "openai" represents the default CLI argument passed through
```
Comment (JuHyung-Son) [inline pkg/analysis/analysis.go:127]: comments need to be fixed

File: pkg/server/analyze.go [hunk 2]
```diff
@@ -16,10 +16,6 @@ func (h *handler) Analyze(ctx context.Context, i *schemav1.AnalyzeRequest) (
 		i.Output = "json"
 	}
 
-	if i.Backend == "" {
```
Comment (JuHyung-Son) [inline pkg/server/analyze.go:19]: why does this lines are removed?

**Review by JuHyung-Son — COMMENTED** [review]
thanks, cli backend options are working well. and i left 2 comments

File: pkg/analysis/analysis.go (hunk 1 above)
Comment (VaibhavMalik4187) [reply pkg/analysis/analysis.go]: Done, thanks for pointing this out.

File: pkg/server/analyze.go (hunk 2 above)
Comment (VaibhavMalik4187) [reply pkg/server/analyze.go:19]: This code was setting the empty backend variable to "openai" (Default value). However, the latest version of the `NewAnalysis` function handles an empty `backend` value in the function itself. Hence, I removed this code.

**Comment by VaibhavMalik4187** [comment]
> This looks like it's just moving the functionality elsewhere?

Consider the following scenario if this fix was not present:
1. User sets the default provider to say `localai`
2. Suppose, I've also configured an "openai" backend and instead of using the default provider, I want to use `openai` to `analyze` the cluster.
3. How would I do that by specifying the `backend` flag? These conditions would enter the following if block and force set the backend to `localai`:
```
if configAI.DefaultProvider != "" && backend == "openai" {
```

However, using the fix I proposed, the user can also specify `openai` as the backend for the `analyze` command. And setting the default value of the `backend` flag as `""` helps in identifying if the user wanted to `openai` or not.

**Comment by JuHyung-Son** [comment]
i think i looks good. and works well.
and im not sure this problem is related to this pr. doe `k8sgpt serve` works??

CleanShot 2024-02-19 at 11 10 47

**Comment by VaibhavMalik4187** [comment]
> i think i looks good. and works well. and im not sure this problem is related to this pr. doe `k8sgpt serve` works??
> 
> CleanShot 2024-02-19 at 11 10 47

I think you hunch is right, this is a separate problem because I saw changes related to `cmd/serve/serve.go` when rebasing https://github.com/k8sgpt-ai/k8sgpt/pull/929 on main. I'll look into it and open a PR with the fix.

**Review by JuHyung-Son — APPROVED** [review]
sorry for delay, this looks good to me.
and cli works well

**Comment by VaibhavMalik4187** [comment]
> sorry for delay, this looks good to me. and cli works well

No worries, I've rebased the PR to included the latest changes on main.

**Pull request merged**